pkg/k8s/apis/cilium.io/v2/register.go

```diff
@@ -548,9 +548,37 @@ var (
 					"AWS": AWSGroup,
 				},
 			},
+			"toFQDNs": {
+				Description: `ToFQDNs is a list of rules matching fqdns that endpoint
+				is allowed to communicate with`,
+				Type: "array",
+				Items: &apiextensionsv1beta1.JSONSchemaPropsOrArray{
+					Schema: &FQDNRule,
+				},
+			},
 		},
 	}
 
+	FQDNRule = apiextensionsv1beta1.JSONSchemaProps{
+		Description: `FQDNRule is a rule that specifies an fully qualified domain name to which outside communication is allowed`,
+		Properties: map[string]apiextensionsv1beta1.JSONSchemaProps{
+			"matchName":    MatchFQDNName,
+			"matchPattern": MatchFQDNPattern,
+		},
+	}
+
+	MatchFQDNName = apiextensionsv1beta1.JSONSchemaProps{
+		Description: `MatchName matches fqdn name`,
+		Type:        "string",
+		Pattern:     `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`,
```

Comment:
I think both of these need a `\.?` at the end, they don't allow `foo.com.` type names.

Reply:
@nebril As this is a regex can you create unit tests for the regex itself and make the regex as a constant in this file?

Reply:
done